Source/swiftlint/Extensions/Array+SwiftLint.swift

```diff
@@ -6,12 +6,14 @@ extension Array {
     }
 
     func parallelMap<T>(transform: (Element) -> T) -> [T] {
-        var result = ContiguousArray<T?>(repeating: nil, count: count)
-        return result.withUnsafeMutableBufferPointer { buffer in
-            DispatchQueue.concurrentPerform(iterations: buffer.count) { idx in
-                buffer[idx] = transform(self[idx])
+        return [T](unsafeUninitializedCapacity: count) { buffer, initializedCount in
+            let baseAddress = buffer.baseAddress!
+            DispatchQueue.concurrentPerform(iterations: count) { index in
+                // Using buffer[index] does assignWithTake which tries
+                // to read the uninitialized value (to release it) and crashes
+                (baseAddress + index).initialize(to: transform(self[index]))
```

Comment:
How does this work? I’d expect to have to multiply the index by the size of the transformed result type.

Reply:
It used the type of the pointer to know how much to increase by. The `+` operator isn't documented, but it looks like this new function is and is documented to behave the same way: https://developer.apple.com/documentation/swift/unsafemutablepointer/3393143-advanced